### ISSUE TITLE
(Automatic) creation of Amazon SQS Queue is broken

### ISSUE THREAD

**Issue by holtkamp** [bernardphp/bernard#294]
https://github.com/bernardphp/bernard/pull/289#issuecomment-279327331
>That's because resolveUrl will throw \InvalidArgumentException when it can't resolve queue url.

But ```SqsDriver::resolveUrl()``` will throw an **SqsException** when the Queue does NOT exist, this is triggered by ```$this->sqs->getQueueUrl(['QueueName' => $queueName]);```:

```
Error executing "GetQueueUrl" on "https://sqs.eu-west-1.amazonaws.com"; AWS HTTP error: Client error: `POST https://sqs.eu-west-1.amazonaws.com` resulted in a `400 Bad Request` response: SenderA (truncated...) AWS.SimpleQueueService.NonExistentQueue (client): The specified queue does not exist for this wsdl version. - SenderAWS.SimpleQueueService.NonExistentQueue
```

This type of Exception is not caught in ```SqsDriver::createQueue()```, so *automatic* creation of a Queue is now broken... I would suggest to have ```SqsDriver::resolveUrl()``` explicitly throw an **SqsException** instead of an \InvalidArgumentException.

**Comment by elazar** [comment]
Was this resolved by #295 / #327?

**Comment by holtkamp** [comment]
@elazar yes! You are right, this issue can be closed.
